routes/render.go

```diff
@@ -57,15 +56,28 @@ func getDarkModeMediaQuery(theme string) string {
 	return "(prefers-color-scheme:dark)"
 }
 
-func getThemeCSS(theme string) template.CSS {
+// If theme is "dark" or "light", only that CSS file is loaded as you expect
+// If theme is auto, then the light theme is loaded, and the dark theme might
+// be loaded based on a <link media="..."> query
+func getThemeCSSLink(theme string) string {
 	switch theme {
 	case "dark":
-		return css["dark"]
+		return assets["css/common/dark.css"]
 	case "light":
-		return css["light"]
+		return assets["css/common/light.css"]
 	}
+	return assets["css/common/light.css"]
+}
 
-	return css["light"] + "@media(prefers-color-scheme:dark){" + css["dark"] + "}"
+// The dark theme is loaded on auto theme in a <link media="..."> query
+func getAutoDarkCSSLink(theme string) string {
+	switch theme {
+	case "dark":
```

Comment:
this would be marginally if the case statements were combined. Or perhaps just invert and use an if
```go
if theme == "auto" {
    return assets["css/common/dark.css"]
}
return ""
```